server/channels/app/metrics.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+)
+
+func (a *App) RegisterPerformanceReport(rctx request.CTX, report *model.PerformanceReport) *model.AppError {
+	if a.Metrics() == nil {
+		return nil
+	}
+
+	commonLabels := report.ProcessLabels()
+
+	for _, c := range report.Counters {
+		switch c.Metric {
+		case model.ClientLongTasks:
+			a.Metrics().IncrementClientLongTasks(commonLabels["platform"], commonLabels["agent"], float64(c.Value))
+		default:
+			// we intentionally skip unknown metrics
```

Comment:
Actually I started with logging and then removed it because I think if there is a case (ie. server receives unknown samples) that would pollute the entire log file quite a bit. And I suspect that this log to be an actionable item by an administrator.